OmniBLE/Bluetooth/PeripheralManager.swift

```diff
@@ -460,17 +506,34 @@ extension PeripheralManager: CBPeripheralDelegate {
 
 extension PeripheralManager: CBCentralManagerDelegate {
     func centralManagerDidUpdateState(_ central: CBCentralManager) {
+        self.log.debug("PeripheralManager - centralManagerDidUpdateState: %@", central)
         switch central.state {
         case .poweredOn:
-            assertConfiguration()
+            self.log.debug("PeripheralManager - centralManagerDidUpdateState - running assertConfiguration")
+            if peripheral.state != .connected {
+                sessionQueue.isSuspended = true
+            }
+            if peripheral.state == .connected {
+                sessionQueue.isSuspended = false
+                assertConfiguration()
+            }
         default:
             break
         }
     }
 
+    func centralManager(_ central: CBCentralManager, didFailToConnect peripheral: CBPeripheral, error: Error?) {
+        // Clear the queue in case of connection error
+        sessionQueue.cancelAllOperations()
```

Comment:
This actually doesn't delete operations from the queue...need to figure out smth else there